Updated utility-security-depos...

tekarrawilkinson opened the pull request:
Adding utility-security-depos....

Review: https://brms-simulator-dev.apps.silver.devops.gov.bc.ca/rule/66ff04494e3904e690293094?version=inReview

tekarrawilkinson commented:
Ready for validation testing.

sbuhovac commented:
I did my initial Validation Testing. I have a couple of feedback points below, otherwise, the rule looks good.

- Input 'Deposit Necessary': additional info, when 'i' in circle is clicked on, says "Family unit requires the deposit in order to obtain/retain accommodation." This sounds like regular security deposit so this one needs to be updated (in Klamm, I assume).
- Eligible amount is restricted to $500. This limit is our policy on expense authority and supervisors can approve over $500. How would a superivsor approve over $500 if BRE restricts to $500? Thoughts? @tekarrawilkinson @prchristenson @Iliana777 

@cevenden is away today and still needs to do Validation Testing.

tekarrawilkinson commented:
> I did my initial Validation Testing. I have a couple of feedback points below, otherwise, the rule looks good.
> 
> * Input 'Deposit Necessary': additional info, when 'i' in circle is clicked on, says "Family unit requires the deposit in order to obtain/retain accommodation." This sounds like regular security deposit so this one needs to be updated (in Klamm, I assume).

_We want to re-use inputs when we can, so I can change the language "Family unit requires the deposit in order to secure accommodation or utility." let me know so I can update it_  



> * Eligible amount is restricted to $500. This limit is our policy on expense authority and supervisors can approve over $500. How would a superivsor approve over $500 if BRE restricts to $500? Thoughts? @tekarrawilkinson @prchristenson @Iliana777
> 
> @cevenden is away today and still needs to do Validation Testing.


_I think it makes sense to remove the limit as the policy doesn't indicate it but the expense authority does. It was added into the map as the other deposit rules didn't even have the EA limit. I actually would like some input from @Iliana777 @prchristenson before this change and how we track the future work as the EA intersects with FODIG_

cevenden commented:
> > I did my initial Validation Testing. I have a couple of feedback points below, otherwise, the rule looks good.
> > 
> > * Input 'Deposit Necessary': additional info, when 'i' in circle is clicked on, says "Family unit requires the deposit in order to obtain/retain accommodation." This sounds like regular security deposit so this one needs to be updated (in Klamm, I assume).
> 
> _We want to re-use inputs when we can, so I can change the language "Family unit requires the deposit in order to secure accommodation or utility." let me know so I can update it_
> 
> > * Eligible amount is restricted to $500. This limit is our policy on expense authority and supervisors can approve over $500. How would a superivsor approve over $500 if BRE restricts to $500? Thoughts? @tekarrawilkinson @prchristenson @Iliana777
> > 
> > @cevenden is away today and still needs to do Validation Testing.
> 
> _I think it makes sense to remove the limit as the policy doesn't indicate it but the expense authority does. It was added into the map as the other deposit rules didn't even have the EA limit. I actually would like some input from @Iliana777 @prchristenson before this change and how we track the future work as the EA intersects with FODIG_

@tekarrawilkinson , I am good with a more generic info bubble and your recommendation works for me.  Inputs otherwise look good.  Is it possible to add an input if the requested amount is above EAW $500 limit asking if Supervisor or Manager approve the higher amount?

I am also noticing there aren't any reasons provided when I run an ineligible test, I only get a FALSE, no reason. Will that be added before Isolation testing?  @sbuhovac @tekarrawilkinson @prchristenson @Iliana777.

prchristenson commented:
> > I did my initial Validation Testing. I have a couple of feedback points below, otherwise, the rule looks good.
> > 
> > * Input 'Deposit Necessary': additional info, when 'i' in circle is clicked on, says "Family unit requires the deposit in order to obtain/retain accommodation." This sounds like regular security deposit so this one needs to be updated (in Klamm, I assume).
> 
> _We want to re-use inputs when we can, so I can change the language "Family unit requires the deposit in order to secure accommodation or utility." let me know so I can update it_
> 
> > * Eligible amount is restricted to $500. This limit is our policy on expense authority and supervisors can approve over $500. How would a superivsor approve over $500 if BRE restricts to $500? Thoughts? @tekarrawilkinson @prchristenson @Iliana777
> > 
> > @cevenden is away today and still needs to do Validation Testing.
> 
> _I think it makes sense to remove the limit as the policy doesn't indicate it but the expense authority does. It was added into the map as the other deposit rules didn't even have the EA limit. I actually would like some input from @Iliana777 @prchristenson before this change and how we track the future work as the EA intersects with FODIG_

Adding my thoughts on the 2 above items:

1. I think this should be a unique input 'Utility Deposit Necessary' that is described as "Family unit requires deposit in order to obtain or continue utility services".  The other input could remain as it is and be used for security and pet damage deposits.

2. I agree with Tekarra that we should remove the $ limit in this rule.  It will need to be part of the Expense Authority parent rule set which is under User Story 6068.   I've added this to the future work table in the Utility Security Deposit Hive Rule Page.

Edited comment to add tags @tekarrawilkinson @sbuhovac @cevenden @Iliana777

tekarrawilkinson commented:
@sbuhovac @cevenden @prchristenson @Iliana777 

I've separated out the utility deposit necessary as its own input because the language is different and updated the scenarios.

The $500 limit has been removed and there is no limit now. I've removed the scenarios referencing this and resaved so they don't mess up the scenario results.

As far as reasons are concerned, I'm still not set on if or how to use them. they are easy to add to the rules, I changed it to one word reasons, see if that's easier for errors? let me know if this helps

cevenden commented:
> @sbuhovac @cevenden @prchristenson @Iliana777
> 
> I've separated out the utility deposit necessary as its own input because the language is different and updated the scenarios.
> 
> The $500 limit has been removed and there is no limit now. I've removed the scenarios referencing this and resaved so they don't mess up the scenario results.
> 
> As far as reasons are concerned, I'm still not set on if or how to use them. they are easy to add to the rules, I changed it to one word reasons, see if that's easier for errors? let me know if this helps

Based on the changes noted, I am happy to pass Utility Security Deposit Validation.
@sbuhovac @tekarrawilkinson @prchristenson @Iliana777

sbuhovac commented:
I reviewed the updated rule, found no further issues, and passed Validation Testing.

Over to @prchristenson and @Iliana777 for Validation Testing.

Iliana777 reviewed (approved):
@tekarrawilkinson @prchristenson @sbuhovac @cevenden Validation test completed. Ran two tests (deposit not necessary and deposit amount in dollars/cents). Both passed as expected. Approving this draft.

prchristenson commented:
@tekarrawilkinson In the policy there is a requirement to seek supervisor approval for a second security deposit to the **same** utility if the person has not yet repaid the first deposit.  This was noted in the map but currently isn't an input in the rule.  cc: @sbuhovac @cevenden @Iliana777

tekarrawilkinson commented:
> @tekarrawilkinson In the policy there is a requirement to seek supervisor approval for a second security deposit to the **same** utility if the person has not yet repaid the first deposit. This was noted in the map but currently isn't an input in the rule. cc: @sbuhovac @cevenden @Iliana777

I've added a simple true/false input. The scenarios will need to be re-saved with the new input, I'll start working no this now in the background, but you are free to re-test or add new scenarios.

prchristenson reviewed (approved):
@tekarrawilkinson Validation Testing Completed.  I added two new scenarios and tested the new input of "has UnpaidDeposit" which both passed.   We note that to receive an eligible output related to issuing a 2nd deposit for the same utility, we need to build "parent rule: overrides not in authority matrix" that would allow for supervisor's approval, when applicable.   @sbuhovac @cevenden - This rule is ready for isolation testing scheduled for Sprint 18.

sbuhovac commented:
> @tekarrawilkinson Validation Testing Completed. I added two new scenarios and tested the new input of "has UnpaidDeposit" which both passed. We note that to receive an eligible output related to issuing a 2nd deposit for the same utility, we need to build "parent rule: overrides not in authority matrix" that would allow for supervisor's approval, when applicable. @sbuhovac @cevenden - This rule is ready for isolation testing scheduled for Sprint 18.

@prchristenson and @Iliana777 , can we discuss this in a quick meeting please? Cris and I reviewed the rule as part of Isolation Testing, and we find the output reason "Unpaid Deposit" doesn't have justification in the Regs. We understand this is our policy (and handled as a business process) but if we want to build Reasons in BRE they would need to be linked to sections of Regs, and there is none for this output.

prchristenson commented:
> > @tekarrawilkinson Validation Testing Completed. I added two new scenarios and tested the new input of "has UnpaidDeposit" which both passed. We note that to receive an eligible output related to issuing a 2nd deposit for the same utility, we need to build "parent rule: overrides not in authority matrix" that would allow for supervisor's approval, when applicable. @sbuhovac @cevenden - This rule is ready for isolation testing scheduled for Sprint 18.
> 
> @prchristenson and @Iliana777 , can we discuss this in a quick meeting please? Cris and I reviewed the rule as part of Isolation Testing, and we find the output reason "Unpaid Deposit" doesn't have justification in the Regs. We understand this is our policy (and handled as a business process) but if we want to build Reasons in BRE they would need to be linked to sections of Regs, and there is none for this output.

@sbuhovac @cevenden - I agree that the regs do not support the denial for a 2nd util sec dep to the same provider; however, policy and process requires the supervisor to approve the 2nd one before it can be issued.  We've flagged that as future work needed so it's not built into this rule.   Until that supervisor approval parent rule is built, the output for "hasUnpaidDeposit" will be a denial.  @tekarrawilkinson @Iliana777  Please comment if I've missed something or I'm incorrect.

Iliana777 commented:
> > > @tekarrawilkinson Validation Testing Completed. I added two new scenarios and tested the new input of "has UnpaidDeposit" which both passed. We note that to receive an eligible output related to issuing a 2nd deposit for the same utility, we need to build "parent rule: overrides not in authority matrix" that would allow for supervisor's approval, when applicable. @sbuhovac @cevenden - This rule is ready for isolation testing scheduled for Sprint 18.
> > 
> > 
> > @prchristenson and @Iliana777 , can we discuss this in a quick meeting please? Cris and I reviewed the rule as part of Isolation Testing, and we find the output reason "Unpaid Deposit" doesn't have justification in the Regs. We understand this is our policy (and handled as a business process) but if we want to build Reasons in BRE they would need to be linked to sections of Regs, and there is none for this output.
> 
> @sbuhovac @cevenden - I agree that the regs do not support the denial for a 2nd util sec dep to the same provider; however, policy and process requires the supervisor to approve the 2nd one before it can be issued. We've flagged that as future work needed so it's not built into this rule. Until that supervisor approval parent rule is built, the output for "hasUnpaidDeposit" will be a denial. @tekarrawilkinson @Iliana777 Please comment if I've missed something or I'm incorrect.

@sbuhovac @cevenden @prchristenson @tekarrawilkinson Is this not similar to the Confirmed Job Supplement rule, where we allowed the max supplement of $1000 to be approved even though the EAW authority is $250? We will need a utility rule for spending authority. Could we not do the same for this rule?

prchristenson commented:
@tekarrawilkinson The tester group met and the consensus we reached was to remove the "hasUnpaidDeposit" input from the Utility Security Deposit rule and move it under a parent supervisor override rule.  This would be consistent with how we are handling other supervisor overrides that are needed for other supplements.   Let us know if you have any concerns or questions about this change.   cc: @sbuhovac @cevenden @Iliana777

prchristenson commented:
@tekarrawilkinson The tester group met and the consensus we reached was to remove the "hasUnpaidDeposit" input from the Utility Security Deposit rule and move it under a parent supervisor override rule.  This would be consistent with how we are handling other supervisor overrides that are needed for other supplements.   Let us know if you have any concerns or questions about this change.   cc: @sbuhovac @cevenden @Iliana777

tekarrawilkinson commented:
@timwekkenbc the CSV rule tests failed on this PR, can you look into this? I want to confirm if this was anything to do with the removal of an input?

timwekkenbc commented:
> @timwekkenbc the CSV rule tests failed on this PR, can you look into this? I want to confirm if this was anything to do with the removal of an input?

Yeah, I'll try to fix this up today. It looks like its just a bug in the new functionality I pushed.

sbuhovac commented:
Isolation Testing shows "No Data" output when input "Utility Deposit Necessary" is FALSE and the other inputs are TRUE. See saved scenarios "Deposit not necessary" and "No Data Result" as reference points.

@tekarrawilkinson

tekarrawilkinson commented:
> Isolation Testing shows "No Data" output when input "Utility Deposit Necessary" is FALSE and the other inputs are TRUE. See saved scenarios "Deposit not necessary" and "No Data Result" as reference points.
> 
@sbuhovac 

the line for a false result was removed at some point. added it back, should give you the false result.

sbuhovac reviewed (approved):
Validation Testing completed, all tests passed.
[general-supplements_utility-security-deposit_testing_minValue_0_maxValue_3000utility-security-depos..._2025-02-26T00-20-29Z.csv](https://github.com/user-attachments/files/18974561/general-supplements_utility-security-deposit_testing_minValue_0_maxValue_3000utility-security-depos._2025-02-26T00-20-29Z.csv)

cevenden reviewed (approved):
Isolation testing approved

timwekkenbc commented:
I fixed the csv testing issues. Hopefully there are no more.
